src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -154,31 +158,43 @@ public function testCopyWithOverrideWithReadOnlyTargetFails()
         $this->filesystem->copy($sourceFilePath, $targetFilePath, true);
     }
 
-    public function testCopyCreatesTargetDirectoryIfItDoesNotExist()
+    /**
+     * @dataProvider provideSourceTargetPaths
+     */
+    public function testCopyCreatesTargetDirectoryIfItDoesNotExist($sourceFilePath, $targetFilePath)
     {
-        $sourceFilePath = $this->workspace.DIRECTORY_SEPARATOR.'copy_source_file';
-        $targetFileDirectory = $this->workspace.DIRECTORY_SEPARATOR.'directory';
-        $targetFilePath = $targetFileDirectory.DIRECTORY_SEPARATOR.'copy_target_file';
+        $targetFileDirectory = dirname($sourceFilePath).DIRECTORY_SEPARATOR.'directory';
+        if ($targetFilePath instanceof StringishObject) {
+            $targetFilePath = new StringishObject($targetFileDirectory.DIRECTORY_SEPARATOR.'copy_target_file');
+        } else {
+            $targetFilePath = $targetFileDirectory.DIRECTORY_SEPARATOR.'copy_target_file';
+        }
 
         file_put_contents($sourceFilePath, 'SOURCE FILE');
 
         $this->filesystem->copy($sourceFilePath, $targetFilePath);
 
         $this->assertTrue(is_dir($targetFileDirectory));
-        $this->assertFileExists($targetFilePath);
+        $this->assertFileExists((string) $targetFilePath);
         $this->assertEquals('SOURCE FILE', file_get_contents($targetFilePath));
     }
 
-    public function testCopyForOriginUrlsAndExistingLocalFileDefaultsToCopy()
+    /**
+     * @dataProvider provideSourceTargetPaths
+     */
+    public function testCopyForOriginUrlsAndExistingLocalFileDefaultsToCopy($sourceFilePath, $targetFilePath)
     {
-        $sourceFilePath = 'http://symfony.com/images/common/logo/logo_symfony_header.png';
-        $targetFilePath = $this->workspace.DIRECTORY_SEPARATOR.'copy_target_file';
+        if ($sourceFilePath instanceof StringishObject) {
+            $sourceFilePath = new StringishObject('http://symfony.com/images/common/logo/logo_symfony_header.png');
+        } else {
+            $sourceFilePath = 'http://symfony.com/images/common/logo/logo_symfony_header.png';
+        }
 
         file_put_contents($targetFilePath, 'TARGET FILE');
 
         $this->filesystem->copy($sourceFilePath, $targetFilePath, false);
```

Comment:
Travis fails on data set 1 and 2 (v2 and v3).. it works on my machine though :confused: 

```
$ php -r "var_dump(stream_is_local('http://symfony.com/images/common/logo/logo_symfony_header.png'), is_file('http://symfony.com/images/common/logo/logo_symfony_header.png'));"
Command line code:1:
bool(false)
Command line code:1:
bool(false)
$ php -v
PHP 5.6.21-1+donate.sury.org~trusty+2 (cli)
```